controller/backup_target_controller.go

```diff
@@ -414,6 +427,67 @@ func (btc *BackupTargetController) reconcile(name string) (err error) {
 	return nil
 }
 
+func (btc *BackupTargetController) syncSystemBackup(backupTargetClient *engineapi.BackupTargetClient, log logrus.FieldLogger) error {
+	backupstoreSystemBackups, err := backupTargetClient.SystemBackupList(backupTargetClient.URL, backupTargetClient.Credential)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list system backups in %v", backupTargetClient.URL)
+	}
+
+	clusterSystemBackups, err := btc.ds.ListSystemBackups()
+	if err != nil {
+		return errors.Wrap(err, "failed to list SystemBackups")
+	}
+
+	clusterReadySystemBackupNames := sets.NewString()
+	for _, systemBackup := range clusterSystemBackups {
+		if systemBackup.Status.State != longhorn.SystemBackupStateReady {
```

Comment:
Filtering out the SystemBackup that is not in state Ready to avoid deleting the one still in-progress or in an Error state.

Reply:
But you will re-create those still not-ready items again, right? even though the creation will be failed later because they already exist in the cluster.